pkg/controller/ibmblockcsi/ibmblockcsi_controller.go

```diff
@@ -369,88 +457,114 @@ func (r *ReconcileIBMBlockCSI) reconcileServiceAccount(instance *ibmblockcsi.IBM
-	externalAttacher := instance.GenerateExternalAttacherClusterRole()
-	externalSnapshotter := instance.GenerateExternalSnapshotterClusterRole()
-	externalResizer := instance.GenerateExternalResizerClusterRole()
-	controllerSCC := instance.GenerateSCCForControllerClusterRole()
-	nodeSCC := instance.GenerateSCCForNodeClusterRole()
+	clusterRoles := r.getClusterRoles(instance)
 
-	for _, cr := range []*rbacv1.ClusterRole{
-		externalProvisioner,
-		externalAttacher,
-		externalSnapshotter,
-		externalResizer,
-		controllerSCC,
-		nodeSCC,
-	} {
-		if err := controllerutil.SetControllerReference(instance.Unwrap(), cr, r.scheme); err != nil {
-			return err
-		}
+	for _, cr := range clusterRoles {
 		found := &rbacv1.ClusterRole{}
 		err := r.client.Get(context.TODO(), types.NamespacedName{
 			Name:      cr.Name,
 			Namespace: cr.Namespace,
 		}, found)
 		if err != nil && errors.IsNotFound(err) {
-			recLogger.Info("Creating a new ClusterRole", "Name", cr.GetName())
+			logger.Info("Creating a new ClusterRole", "Name", cr.GetName())
 			err = r.client.Create(context.TODO(), cr)
 			if err != nil {
 				return err
 			}
 		} else if err != nil {
-			recLogger.Error(err, "Failed to get ClusterRole", "Name", cr.GetName())
+			logger.Error(err, "Failed to get ClusterRole", "Name", cr.GetName())
 			return err
 		} else {
 			// Resource already exists - don't requeue
-			//recLogger.Info("Skip reconcile: ClusterRole already exists", "Name", cr.GetName())
+			//logger.Info("Skip reconcile: ClusterRole already exists", "Name", cr.GetName())
 		}
 	}
 
 	return nil
 }
 
-func (r *ReconcileIBMBlockCSI) reconcileClusterRoleBinding(instance *ibmblockcsi.IBMBlockCSI) error {
-	recLogger := log.WithValues("Resource Type", "ClusterRoleBinding")
+func (r *ReconcileIBMBlockCSI) deleteClusterRolesAndBindings(instance *ibmblockcsi.IBMBlockCSI) error {
+	if err := r.deleteClusterRoleBindings(instance); err != nil {
+		return err
+	}
 
-	externalProvisioner := instance.GenerateExternalProvisionerClusterRoleBinding()
-	externalAttacher := instance.GenerateExternalAttacherClusterRoleBinding()
-	externalSnapshotter := instance.GenerateExternalSnapshotterClusterRoleBinding()
-	externalResizer := instance.GenerateExternalResizerClusterRoleBinding()
-	controllerSCC := instance.GenerateSCCForControllerClusterRoleBinding()
-	nodeSCC := instance.GenerateSCCForNodeClusterRoleBinding()
+	if err := r.deleteClusterRoles(instance); err != nil {
+		return err
+	}
+	return nil
+}
 
-	for _, crb := range []*rbacv1.ClusterRoleBinding{
+func (r *ReconcileIBMBlockCSI) deleteClusterRoles(instance *ibmblockcsi.IBMBlockCSI) error {
+	logger := log.WithName("deleteClusterRoles")
+
+	clusterRoles := r.getClusterRoles(instance)
+
+	for _, cr := range clusterRoles {
+		found := &rbacv1.ClusterRole{}
+		err := r.client.Get(context.TODO(), types.NamespacedName{
+			Name:      cr.Name,
+			Namespace: cr.Namespace,
+		}, found)
+		if err != nil && errors.IsNotFound(err) {
+			continue
```

Comment:
Why don't we want to log the error in case we're not familiar with the error?

Reply:
we are familiar with the error, it is a "not found" error here

Reply:
Ohhh. I thought the error wasn't found :)